### ISSUE TITLE
Unable to push an image to GCR using `docker_registry_image`

### ISSUE THREAD

**Issue by topecongiro** [kreuzwerker/terraform-provider-docker#380]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and docker Provider) Version

```
Terraform v1.2.1
on darwin_amd64
+ provider registry.terraform.io/kreuzwerker/docker v2.16.0
```

### Affected Resource(s)

* `docker_registry_image`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    docker = {
      source  = "kreuzwerker/docker"
      version = "2.16.0"
    }
  }

  backend "gcs" {
    bucket = "terraform-state"
    prefix = "terraform/state"
  }
}

provider "docker" {
  alias = "gcr_provider"

  registry_auth {
    address     = "asia.gcr.io"
    config_file = pathexpand("~/.docker/config.json")
  }
}

resource "docker_registry_image" "image" {
  provider = docker.gcr_provider

  name = "asia.gcr.io/gcp-project/test-image"

  build {
    context = "${path.cwd}/app"
  }
}
```

### Expected Behaviour

Terraform will successfully build and push an image to GCR.

### Actual Behaviour

`terraform apply` fails with the following error:

```
╷
│ Error: Error pushing docker image: Error response from daemon: Bad parameters and missing X-Registry-Auth: EOF
│
│   with docker_registry_image.image,
│   on main.tf line 24, in resource "docker_registry_image" "image":
│   24: resource "docker_registry_image" "image" {
│
╵
```

### Steps to Reproduce

1. `gcloud auth login`
2. `gcloud auth configure-docker`
3. `terraform apply`

### Note

Possible related debug log:
```
2022-05-26T18:54:06.230+0900 [WARN]  Provider "registry.terraform.io/kreuzwerker/docker" produced an invalid plan for docker_registry_image.image, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .insecure_skip_verify: planned value cty.False for a non-computed attribute
      - .keep_remotely: planned value cty.False for a non-computed attribute
      - .build[0].context: planned value cty.StringVal("/Users/seiichi.uchida/dev/terraform/app:99a85bc2868e59ff204700b06055a9af219ffc0fdd93169387f611373fcb4137") does not match config value cty.StringVal("/Users/seiichi.uchida/dev/terraform/app")
      - .build[0].dockerfile: planned value cty.StringVal("Dockerfile") for a non-computed attribute
```

**Comment by envikta** [comment]
in my case, I'm trying to register the image through Github

**Comment by tiagoaragao** [comment]
I'm trying to register the image through Gitlab
Error: Error pushing docker image: Error response from daemon: Bad parameters and missing X-Registry-Auth: EOF

```
provider "docker" { 
  alias = "private"   
  registry_auth {
    address = "us.gcr.io" 
    username = "oauth2accesstoken"
    password = "${data.google_service_account_access_token.default.access_token}"    
  }
}
```

**Comment by Junkern** [comment]
@topecongiro I tried to replicate your issue, but on my local machine it is working wonderful.
I did encounter that issue, though, when the `gcloud` binary was not available on my path, though.

So, I will add some additional warnings, when the credentailshelper cannot provide a proper username/password to the provider.

As with the debug-log: This is also on my todo list.

All problems in this issue some to be working now, will close. Feel free to open a new issue when you encounter a new problem :)
